### PR TITLE
[bug 1167192] Copy Jinja2 i18n extension from tower.

### PR DESCRIPTION
**Pull request by rlr**
I only brought the bits that collapse whitespace in blocks.

r?

### CODE REVIEW

**Comment by willkg** [comment]
This looks awesome. I want to take it through its paces. I'll do that tonight.

File: fjord/base/l10n.py [hunk 1]
```diff
@@ -11,3 +16,19 @@ class Translation(object):
...
+class MozInternationalizationExtension(InternationalizationExtension):
+    """
+    We override jinja2's _parse_block() to collapse whitespace so we can have
+    linebreaks wherever we want.
+    """
+
+    def _parse_block(self, parser, allow_pluralize):
+        parse_block = InternationalizationExtension._parse_block
+        ref, buffer = parse_block(self, parser, allow_pluralize)
+        return ref, strip_whitespace(buffer)
```
Comment (willkg) [inline fjord/base/l10n.py]: Is this better written as:

```
def _parse_block(self, *args, **kwargs):
    ref, buff = super(MozInternationalizationExtension, self)._parse_block(*args, **kwargs)
    return ref, strip_whitespace(buff)
```

?

**Comment by willkg** [comment]
I had that one nit. Otherwise this looks ok. I'll test it out now.

File: fjord/base/l10n.py (hunk 1 above)
Comment (rlr) [inline fjord/base/l10n.py]: That's does look like the right way... I'll change it and make sure stuff still works

**Comment by willkg** [comment]
Looks good testing-wise. I think I'll just land it and we can deal with the nit later if we feel like it.

**Comment by rlr** [comment]
thank you!

**Pull request merged**